### ISSUE TITLE
LMMS uses microphone

### ISSUE THREAD

**Issue by Positron832Alt** [LMMS/lmms#7520]
### System Information

Windows 10 22H2

### LMMS Version(s)

1.3.0-alpha.1.682+g58ce9b476. Still exists in 1.3.0-alpha.1.699+g6a7b23b27.

### Most Recent Working Version

Shortly after LOMM was added was the last version I used without the bug.

### Bug Summary

When opening LMMS the microphone icon in the taskbar appears. Hovering over it displays a tooltip saying LMMS is using the microphone. Note that I use a laptop.

### Expected Behaviour

LMMS should not use the microphone.

### Steps To Reproduce

1. Open LMMS
2. Observe the microphone icon in the taskbar appear

### Logs

No logs.

### Please search the issue tracker for existing bug reports before submitting your own.

- [X] I have searched all existing issues and confirmed that this is not a duplicate.

**Comment by michaelgregorius** [comment]
@Positron832Alt, can you please check the "Audio" section in the settings ("Edit > Settings" in the main menu) and report what driver you are using? I assume that it will be SDL and look somewhat like as follows:

![7520-AudioConfiguration](https://github.com/user-attachments/assets/d8783819-82d6-487e-952a-92055b5e4d33)

Can you then please check what options are shown for you under "Input device"? I assume that the "[System default]" on your system will be the microphone. If there are several options available in the combo box you might be able switch to an input which is not the microphone. As far as I know it is not possible to select something like "None" for the SDL driver.

**Comment by michaelgregorius** [comment]
Because it's the only option it's also the system default. Unfortunately, it will therefore always be used as the input device because there is no "None" option.

**Comment by michaelgregorius** [comment]
The SDL input feature seems to be fairly new. I have just checked and LMMS 1.2.2 does not have it.

Are you sure that it does not happen with a version from shortly after LOMM has been introduced? LOMM has been introduced to the code base on November 10th, 2023 (see 89c98a77a51).

Here's the history of the SDL driver implementation: https://github.com/LMMS/lmms/commits/master/src/core/audio/AudioSdl.cpp

If I check a version of the SDL driver from before November 10th, e.g. https://github.com/LMMS/lmms/blob/3aed361b82dba6246119c8fc9d929fd9a779a647/src/core/audio/AudioSdl.cpp, then it also tries to open the default input device. So the usage of the microphone should also be indicated with that version.

I also wonder if the issue should really be closed. It might be used to discuss if there should be an option to disable audio input in the driver(s).

**Comment by musikBear** [comment]
> discuss if there should be an option to disable audio input in the driver(s).

Indeed! And that should be _yes_ ofcause. LMMS cant record, so resources used by microphone is wasted

**Comment by firewall1110** [comment]
If enable input device only if needed (as it was in #6070) , that 
- should fix this bug;
- make possible change input device without LMMS restart; 

P.S.
May be using this bugfix context I will make one (not enabled in #5990) recording button working (to simply record to file <project name><some number>.wav), but with simple - not Real Time Safe way.

**Comment by firewall1110** [comment]
It is **_not only windows specific_** but most Linux desktop have no _microphone icon in the taskbar_ , so user can not so easy observe the fact (lmms with SDL driver uses microphone). 

P.S.
make one recording button working was much easy task for me then fix this bug (if it is bug): https://github.com/LMMS/lmms/pull/6070 solution is not enough, **need** even **not init input** on start but only then needed , so work is in progress.
